File: perf/bench/collatz/collatz.go
package main

import "fmt"

func steps(n int64) int64 {
	var s int64 = 0
	for n != 1 {
		if n%2 == 0 {
			n = n / 2
		} else {
			n = 3*n + 1
		}
		s++
	}
	return s
}

func main() {
	var total int64 = 0
	for i := int64(1); i < 5000000; i++ {
		total += steps(i)
	}
	fmt.Println(total)
}

File: perf/bench/collatz/collatz.swift
func steps(_ n0: Int64) -> Int64 {
    var n = n0
    var s: Int64 = 0
    while n != 1 {
        if n % 2 == 0 { n = n / 2 } else { n = 3 &* n &+ 1 }
        s += 1
    }
    return s
}
var total: Int64 = 0
var i: Int64 = 1
while i < 5000000 {
    total += steps(i)
    i += 1
}
print(total)
